src/ring_theory/ideal/basic.lean

```diff
@@ -490,16 +540,38 @@ namespace ideal
 /-- All ideals in a division ring are trivial. -/
 lemma eq_bot_or_top : I = ⊥ ∨ I = ⊤ :=
 begin
-  rw or_iff_not_imp_right,
-  change _ ≠ _ → _,
-  rw ideal.ne_top_iff_one,
+  rw [or_iff_not_imp_right, ← ne.def, ideal.ne_top_iff_one],
   intro h1,
   rw eq_bot_iff,
   intros r hr,
   by_cases H : r = 0, {simpa},
   simpa [H, h1] using I.mul_mem_left r⁻¹ hr,
 end
 
+@[priority 100] -- see Note [lower instance priority]
+instance : fintype $ ideal K :=
+{ elems := finset.cons ⊥ {⊤} $ finset.mem_singleton.not.mpr bot_ne_top,
+  complete := λ I, by simp [eq_bot_or_top], }
+
+@[simp] lemma card_ideal : fintype.card (ideal K) = 2 := rfl
+
+lemma _root_.right_ideal.eq_bot_or_top (I : right_ideal K) : I = ⊥ ∨ I = ⊤ :=
+begin
+  rw [or_iff_not_imp_right, ← ne.def, right_ideal.ne_top_iff_one],
+  intros h1,
+  rw eq_bot_iff,
+  intros r hr,
+  by_cases H : r = 0, {simpa},
+  simpa [H, h1] using I.mul_mem_right r⁻¹ hr,
+end
+
+@[priority 100] -- see Note [lower instance priority]
+instance : fintype $ right_ideal K :=
```

Comment:
This instance is going to end up with some weird name like `ideal.right_ideal.fintype`. Can you move it out of the `ideal` namespace?

Reply:
Done.